README.md

```diff
@@ -1,15 +1,107 @@
+get_numeric_assignment(...)
+get_integer_assignment(...)
+get_string_assignment(...)
+get_json_assignment(...)
+```
+
+Each function has the same signature, but returns the type in the function name. For booleans use `get_boolean_assignment`, which has the following signature:
+
+```python
+get_boolean_assignment(
+    flag_key: str,
+    subject_key: str,
+    subject_attributes: Dict[str, Any],
+    default_value: bool
+) -> bool:
+  ```
+
+## Initialization options
+
+The `init` function accepts the following optional configuration arguments.
+
+| Option | Type | Description | Default |
+| ------ | ----- | ----- | ----- |
+| **`assignment_logger`**  | [AssignmentLogger](https://github.com/Eppo-exp/python-sdk/blob/ebc1a0b781769fe9d2e2be6fc81779eb8685a6c7/eppo_client/assignment_logger.py#L6-L10) | A callback that sends each assignment to your data warehouse. Required only for experiment analysis. See [example](#assignment-logger) below. | `None` |
+| **`is_graceful_mode`** | bool | When true, gracefully handles all exceptions within the assignment function and returns the default value. | `True` |
+
+
+## Assignment logger
+
+To use the Eppo SDK for experiments that require analysis, pass in a callback logging function to the `init` function on SDK initialization. The SDK invokes the callback to capture assignment data whenever a variation is assigned. The assignment data is needed in the warehouse to perform analysis.
+
+The code below illustrates an example implementation of a logging callback using [Segment](https://segment.com/), but you can use any system you'd like. The only requirement is that the SDK receives a `log_assignment` callback function. Here we define an implementation of the Eppo `SegmentAssignmentLogger` interface containing a single function named `log_assignment`:
+
+```python
+from eppo_client.assignment_logger import AssignmentLogger
+import analytics
+
+# Connect to Segment.
+analytics.write_key = "<SEGMENT_WRITE_KEY>"
+
+class SegmentAssignmentLogger(AssignmentLogger):
+    def log_assignment(self, assignment):
+        analytics.track(assignment["subject"], "Eppo Randomization Assignment", assignment)
+
+client_config = Config(api_key="<SDK-KEY-FROM-DASHBOARD>", assignment_logger=SegmentAssignmentLogger())
+```
+
+## Philosophy
+
+Eppo's SDKs are built for simplicity, speed and reliability. Flag configurations are compressed and distributed over a global CDN (Fastly), typically reaching your servers in under 15ms. Server SDKs continue polling Eppo’s API at 30-second intervals. Configurations are then cached locally, ensuring that each assignment is made instantly. Each SDK is as light as possible, with evaluation logic at around [25 simple lines of code](https://github.com/Eppo-exp/python-sdk/blob/ebc1a0b781769fe9d2e2be6fc81779eb8685a6c7/eppo_client/eval.py#L39-L69). The simple typed functions listed above are all developers need to know about, abstracting away the complexity of the underlying set of features.
```

Comment:
Same comment on omitting lines of code and just highlighting it does simple string and numeric comparisons